### PR TITLE
[ENH] Fix _update from STLForecaster to handle update_params=False

### PR DESCRIPTION
**Pull request by aiwalter**
##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.
- [ ] The PR title starts with either [ENH], [DOC] or [BUG] indicating whether the PR topic is related to enhancement, documentation or bug

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by aiwalter** [comment]
@fkiraly seems the PR #2466 did not resolve the issue?

**Comment by fkiraly** [comment]
This should do it!
https://github.com/alan-turing-institute/sktime/pull/2557

File: extension_templates/forecasting.py [hunk 1]
```diff
@@ -22,9 +22,12 @@
 Mandatory implements:
     fitting         - _fit(self, y, X=None, fh=None)
     forecasting     - _predict(self, fh=None, X=None)
+    updating        - _update(self, y, X=None, update_params=True).
+                    - Only mandatory to for compositions, see e.g.  STLForecaster
```
Comment (fkiraly) [inline extension_templates/forecasting.py:26]: I think this is misleading, people would ready this as "not mandatory if not a composition", or not know what a composition is. So I would recommend to remove this.
I would add explanatory text in the body instead, about `update_params=False`.

File: sktime/forecasting/trend.py [hunk 2]
```diff
@@ -452,28 +452,42 @@ def _update(self, y, X=None, update_params=True):
...
+                seasonal_deg=self.seasonal_deg,
+                trend_deg=self.trend_deg,
+                low_pass_deg=self.low_pass_deg,
+                robust=self.robust,
+                seasonal_jump=self.seasonal_jump,
+                trend_jump=self.trend_jump,
+                low_pass_jump=self.low_pass_jump,
+            ).fit(inner_iter=self.inner_iter, outer_iter=self.outer_iter)
+
+            self.seasonal_ = pd.Series(self._stl.seasonal, index=y.index)
```
Comment (fkiraly) [inline sktime/forecasting/trend.py:480]: seems incorrect to me, because `y` is only the new data, not all data

**Review by fkiraly — CHANGES_REQUESTED** [review]
Yes, this is the right fix imo!

I would kindly ask you to change the comments in the extension template - details like that should not be in the table of contents, but in the explanation of what to fill in in `_update`.

**Comment by fkiraly** [comment]
this can't be replaced by the default because we have three components, which need to be updated by specific `y` (not by the one passed to `update`.

**Pull request closed without merging**